api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java

```diff
@@ -0,0 +1,103 @@
+package org.openmrs.module.eptsreports.reporting.library.cohorts;
+
+import java.util.*;
+import org.apache.commons.text.StringSubstitutor;
+import org.openmrs.module.eptsreports.metadata.CommonMetadata;
+import org.openmrs.module.eptsreports.metadata.HivMetadata;
+import org.openmrs.module.eptsreports.reporting.utils.EptsReportUtils;
+import org.openmrs.module.reporting.cohort.definition.*;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class QualityImprovement2020CohortQueries {
+
+  private GenericCohortQueries genericCohortQueries;
+
+  private HivMetadata hivMetadata;
+
+  private CommonMetadata commonMetadata;
+
+  private final String MAPPING = "startDate=${startDate},endDate=${endDate},location=${location}";
```

Comment:
How are the other 2 remaining parameters suppossed to be passed? are they suppossed to persist to the cohorts?

Reply:
it will depend on user stories, according with ticket 679 user story  we only need startdate, enddate  and location for now. we can  add   according with ticket's  needs